python/tritonserver/_api/_request.py

```diff
@@ -167,6 +167,10 @@ def _set_parameters(self, request):
                 request.set_int_parameter(key, value)
             elif isinstance(value, bool):
                 request.set_bool_parameter(key, value)
+            elif isinstance(value, float):
+                request.set_double_parameter(key, value)
+            elif value is None:
```

Comment:
Can you elaborate more about this part and why it was required for OpenAI chat completions?

Reply:
double was to allow for setting temperature in vllm sampling parameters (and should be added in anycase for completeness)

on the None its more related to requests specified in open api specs like:

 https://github.com/triton-inference-server/tutorials/blob/0774e8454f1d6a9ed995e005d99a24004682a1c8/Triton_Inference_Server_Python_API/examples/fastapi/fastapi-codegen/openai_protocol_types.py#L64 

The pydantic type fills in None for values that aren't present so this was mainly for convience of being able to pass through the request parameters.

https://github.com/triton-inference-server/tutorials/blob/0774e8454f1d6a9ed995e005d99a24004682a1c8/Triton_Inference_Server_Python_API/examples/fastapi/fastapi-codegen/main.py#L153

I think it's easy to workaround - but thought its a nice convience to ignore parameters that are None - I'm a little ambivalent to it now.

Reply:
Thanks for explaining. Could we prune these values before passing as request parameters? I also don't have strong opinions. Might be worth mentioning it in the docstrings so that the behaviour is clear to the user.

Reply:
agree we can prune for now - let me take a stab at that and then revert this - on second thought probably better to error cause it may hide some error